apps/prairielearn/assets/scripts/questionsTable.js

```diff
@@ -0,0 +1,190 @@
+        },
+      };
+    }
+
+    $('#questionsTable').bootstrapTable(tableSettings);
+
+    $(document).keydown((event) => {
+      if (
+        (event.ctrlKey || event.metaKey) &&
+        String.fromCharCode(event.which).toLowerCase() === 'f'
+      ) {
+        if ($('.sticky-header-container:visible input.bootstrap-table-filter-control-qid').length) {
+          $('.sticky-header-container:visible input.bootstrap-table-filter-control-qid').focus();
+        } else {
+          $('input.bootstrap-table-filter-control-qid').focus();
+        }
+        event.preventDefault();
+      }
+    });
+  });
+
+  window.topicList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(data, (row) => row.topic.name));
+  };
+
+  window.tagsList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(_.flatten(_.filter(_.map(data, (row) => row.tags))), (row) => row.name));
+  };
+
+  window.versionList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(data, (row) => row.display_type));
+  };
+
+  const urlPrefix = decodeData('url-prefix');
+  window.qidFormatter = function (qid, question) {
+    var text = '';
+    if (question.sync_errors) {
+      text += `<button class="btn btn-xs mr-1" data-toggle="popover" data-title="Sync Errors"
+                        data-content="<pre style=&quot;background-color: black&quot; class=&quot;text-white rounded p-3&quot;>${_.escape(
+                          question.sync_errors_ansified,
+                        )}</pre>">
+            <i class="fa fa-times text-danger" aria-hidden="true"></i>
+            </button>`;
+    } else if (question.sync_warnings) {
+      text += `<button class="btn btn-xs mr-1" data-toggle="popover" data-title="Sync Warnings"
+                    data-content="<pre style=&quot;background-color: black&quot; class=&quot;text-white rounded p-3&quot;>${_.escape(
+                      question.sync_warnings_ansified,
+                    )}</pre>">
+            <i class="fa fa-exclamation-triangle text-warning" aria-hidden="true"></i>
+            </button>`;
+    }
+    text += `<a class="formatter-data" href="${urlPrefix}/question/${question.id}/">${_.escape(
+      question.qid,
+    )}</a>`;
+    if (question.open_issue_count > 0) {
+      text += `<a class="badge badge-pill badge-danger ml-1" href="${urlPrefix}/course_admin/issues?q=is%3Aopen+qid%3A${_.escape(
+        question.qid,
+      )}">${question.open_issue_count}</a>`;
+    }
+    return text;
+  };
+
+  window.topicFormatter = function (topic, question) {
+    return `<span class="badge color-${question.topic.color}">${_.escape(
+      question.topic.name,
+    )}</span>`;
+  };
+
+  window.tagsFormatter = function (tags, question) {
+    return _.map(
+      question.tags ?? [],
+      (tag) => `<span class="badge color-${tag.color}">${tag.name}</span>`,
```

Comment:
Avoid interpolating user-controlled values directly into HTML; this is how you get XSS vulnerabilities. Consider using the `html` tagged template literal from `@prairielearn/html` to generate HTML here and in this file as a whole, as that will automatically escape interpolated values.

Reply:
Yep, I just always use `html` tagged strings for everything and that automatically avoids XSS.

Reply:
done